src/plugins/newtype/README.md

```diff
@@ -0,0 +1,185 @@
+- infos = Information about type plugin is in keys below
+- infos/author = Klemens Böswirth <k.boeswirth+git@gmail.com>
+- infos/licence = BSD
+- infos/provides = check
+- infos/needs =
+- infos/placements = postgetstorage presetstorage
+- infos/status = maintained nodep libc tested unittest recommended
+- infos/metadata = check/type type check/enum check/enum/# check/enum/multi
+- infos/description = type checker using COBRA data types
+
+## Introduction
+
+This plugin is a type checker plugin using the `CORBA` data types.
+
+A common and successful type system happens to be CORBA. The system is well suited because of the many well-defined
+mappings it provides to other programming languages.
+
+The type checker plugin supports these types:
+`short`, `unsigned_short`, `long`, `unsigned_long`, `long_long`, `unsigned_long_long`, `float`, `double`, `char`, `wchar`, `boolean`,
+`any`, `enum`, `string`, `wstring` and `octet`.
+
+- Checking `any` will always be successful, regardless of the content.
+- `string` matches any non-empty key value.
+- `empty` only matches empty key values.
+- `octet` and `char` are equivalent to each other.
+- `enum` will do enum checking as described below.
+- To use `wchar` and `wstring` the function `mbstowcs(3)` must be able convert the key value into a wide character string. `wstring`s can
+  be of any non-zero length, `wchar` must have exactly length 1.
+
+## Enums
+
+If a key is set to the type `enum` the plugin will look for the metadata array `check/enum/#`.
```

Comment:
Do the enums also support normalization? (To the representing int value.)

If not, why including it into one plugin?

Reply:
No that isn't supported. We could implement it, but the ultimate goal should be that applications use the highlevel API with code generation, which solves this problem differently and probably more efficiently than we could inside the plugin.

Its also not really that useful, because the int value will be stored as a string, so instead of doing a few `strcmp` (or nested switches like the highlevel API) you will have to do `strtol`. The only case where this would have significant benefits is when using the highlevel API without code generation.

Reply:
Yes exactly: everything we have in the plugins works for everyone, also those who do not use the high-level API.